series/views.py

```diff
@@ -33,3 +33,16 @@ def get(self, request, term: str):
             request=request,
             context={'shows_list': shows}
         )
+
+
+class ShowDetailsView(View):
+    """Details of a show"""
+    template = 'series/show_details.html'
+
+    def get(self, request, id: int):
+        show = get_show_details(id)
+        return render(
+            template_name='series/show_details.html',
+            request=request,
+            context={'show': show, 'directors': ", ".join(show.directors), 'genres': ", ".join(show.genres)}
```

Comment:
Looks like you're mixing the "logic" and "presentation" layers here… ;-)

A better approach would be to pass `show` object only in the context, and use **template tags**. They allow to perform formatting/transformations on variables in templates.

For example, the `join` template tags does what you do here with `str.join()`:

```
{{ show.directors|join:', ' }}
```

See docs on `join`: https://docs.djangoproject.com/fr/2.1/ref/templates/builtins/#join